src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/MLDsa/MLDsaTests.cs

```diff
@@ -0,0 +1,168 @@
+            byte[] publicKey;
+            byte[] data = [ 1, 2, 3, 4, 5 ];
+            byte[] signature;
+
+            using (MLDsa mldsa = MLDsa.GenerateKey(algorithm))
+            {
+                signature = new byte[algorithm.SignatureSizeInBytes];
+                Assert.Equal(signature.Length, mldsa.SignData(data, signature));
+                Assert.True(mldsa.VerifyData(data, signature));
+
+                publicKey = new byte[algorithm.PublicKeySizeInBytes];
+                Assert.Equal(publicKey.Length, mldsa.ExportMLDsaPublicKey(publicKey));
+            }
+
+            using MLDsa mldsaPub = MLDsa.ImportMLDsaPublicKey(algorithm, publicKey);
+
+            Assert.True(mldsaPub.VerifyData(data, signature));
+            data[0] ^= 1;
+            Assert.False(mldsaPub.VerifyData(data, signature));
+        }
+
+        [ConditionalTheory(typeof(MLDsa), nameof(MLDsa.IsSupported))]
+        [MemberData(nameof(AllMLDsaAlgorithms))]
+        public static void GenerateExportSecretKeySignAndVerify(MLDsaAlgorithm algorithm)
+        {
+            byte[] secretKey;
+            byte[] data = [ 1, 2, 3, 4, 5 ];
+            byte[] signature;
+
+            using (MLDsa mldsaTmp = MLDsa.GenerateKey(algorithm))
+            {
+                signature = new byte[algorithm.SignatureSizeInBytes];
+                Assert.Equal(signature.Length, mldsaTmp.SignData(data, signature));
+
+                secretKey = new byte[algorithm.SecretKeySizeInBytes];
+                Assert.Equal(secretKey.Length, mldsaTmp.ExportMLDsaSecretKey(secretKey));
+            }
+
+            using MLDsa mldsa = MLDsa.ImportMLDsaSecretKey(algorithm, secretKey);
+
+            Assert.True(mldsa.VerifyData(data, signature));
+
+            ((Span<byte>)signature).Fill(0);
+            Assert.Equal(signature.Length, mldsa.SignData(data, signature));
+
+            Assert.True(mldsa.VerifyData(data, signature));
+            data[0] ^= 1;
+            Assert.False(mldsa.VerifyData(data, signature));
+        }
+
+        [ConditionalTheory(typeof(MLDsa), nameof(MLDsa.IsSupported))]
+        [MemberData(nameof(AllMLDsaAlgorithms))]
+        public static void GenerateExportPrivateSeedSignAndVerify(MLDsaAlgorithm algorithm)
+        {
+            byte[] privateSeed;
+            byte[] data = [ 1, 2, 3, 4, 5 ];
+            byte[] signature;
+
+            using (MLDsa mldsaTmp = MLDsa.GenerateKey(algorithm))
+            {
+                signature = new byte[algorithm.SignatureSizeInBytes];
+                Assert.Equal(signature.Length, mldsaTmp.SignData(data, signature));
+
+                privateSeed = new byte[algorithm.PrivateSeedSizeInBytes];
+                Assert.Equal(privateSeed.Length, mldsaTmp.ExportMLDsaPrivateSeed(privateSeed));
+            }
+
+            using MLDsa mldsa = MLDsa.ImportMLDsaPrivateSeed(algorithm, privateSeed);
+
+            Assert.True(mldsa.VerifyData(data, signature));
+
+            ((Span<byte>)signature).Fill(0);
```

Comment:
Why not `signature.AsSpan().Fill(0)`?  I'm not sure I've ever seen this pattern before.

Reply:
too many syntaxes 😄